packages/mgt-components/src/components/mgt-search-box/mgt-search-box.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { html, TemplateResult } from 'lit';
+import { property } from 'lit/decorators.js';
+import { customElement, MgtBaseComponent } from '@microsoft/mgt-element';
+import { fluentSearch } from '@fluentui/web-components/dist/esm/search';
+import { registerFluentComponents } from '../../utils/FluentComponents';
+import { strings } from './strings';
+import { styles } from './mgt-search-box-css';
+import { debounce } from '../../utils/Utils';
+
+registerFluentComponents(fluentSearch);
+
+/**
+ * Web component used to enter a search value to power search scenarios
+ *
+ * @fires {CustomEvent<string>} searchTermChanged - Fired when the search term is changed
+ *
+ * @cssprop --search-input-width - {Length} Search input width
```

Comment:
Genuine question on philosophy of extensibility/styling here. Should we even have this? The element is a 100% wide element, an implementer can easily wrap this in a container that constrains the width better than passing in a width.

Consider setting this to 50%, a common enough scenario. This has the search input set to 50% of the width of the Shadow Root host. i.e. the <mgt-search-box> still takes 100% of its container but the input is constrained to half of that.

By not having the property to adjust here it makes it harder for developers to screw it up

Reply:
Very, very fair! We should get rid of it!